src/Components/Web.Extensions/src/InputFile/FileListEntry.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal class FileListEntry : IFileListEntry
```

Comment:
I'm a little uncertain if we should require users to manually call `IFileListEntry.Dispose()`, since `IFileListEntry` instances are created by us, not the user.

We could `await` the `OnChange` callback and dispose each `IFileListEntry` ourselves, but this prevents users from using `IFileListEntry` instances outside of the `OnChange`, which I don't think is ideal.

I do see a problem with the current API, in that it might be unclear to users whether they should be disposing the `IFileListEntry.Data` stream. As it currently stands, the `Data` property lazily initializes the underlying stream, so users might think they're using a `Stream` created by us (in which case disposing the stream is not their responsibility), but actually, by using the `Data` stream, they are implicitly creating the stream and it is their responsibility to dispose it. I'm not sure this is an implementation detail we should expect users to know.

For clarity, I think it might be worth replacing the `Data` property with an `OpenFileStream` method that always opens a new file stream (i.e. doesn't cache and return a single `Stream`). That way, it should be clear to users that they are responsible for disposing the stream, since they are the ones who created it.

@SteveSandersonMS, do you have any thoughts about this?

Reply:
> For clarity, I think it might be worth replacing the Data property with an OpenFileStream method that always opens a new file stream (i.e. doesn't cache and return a single Stream). That way, it should be clear to users that they are responsible for disposing the stream, since they are the ones who created it.

That's a great idea. I really like the extra clarity this brings. It's much more obvious then that `IFileListEntry` itself doesn't need to be disposable.

On naming, I don't know if we need to call it `OpenFileStream`. It might be clear enough just to call it `Open` as a nod towards [`File.Open`](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.open?view=netcore-3.1). If you find other prior art that makes a particular name seem more consistent/preferable then please go with whatever you like. We'll review this when we get to API review but if you have a justification for whatever you choose it will likely stick :)

Reply:
`IFormFile` has `OpenReadStream` which also indicates that it is read only which also applies here.

https://github.com/dotnet/aspnetcore/blob/3dd75ea18d10320afbdb1c2d6a0782423a62d0a6/src/Http/Http.Features/src/IFormFile.cs#L48